core/jobs/types/base_validation_errors.py

```diff
@@ -64,32 +56,20 @@ def __init__(
         if not message:
             raise ValueError('message must be a non-empty string')
 
-        if (
-            model_id is None and (
-            isinstance(model_or_kind, base_models.BaseModel))
-        ):
-            model_id = job_utils.get_model_id(model_or_kind)
-            model_kind = job_utils.get_model_kind(model_or_kind)
-        elif isinstance(model_or_kind, str):
-            model_kind = model_or_kind
-
-        if model_id:
-            error_message = '%s in %s(id=%s): %s' % (
-                self.__class__.__name__,
-                model_kind,
-                utils.quoted(model_id),
-                message
-            )
-        else:
-            error_message = '%s in %s: %s' % (
-                self.__class__.__name__,
-                model_kind,
-                message
-            )
+        model_id = job_utils.get_model_id(model)
+        if not model_id:
+            model_id = 'missing model Id'
```

Comment:
This seems off to me. Why are we setting model_id to something it shouldn't be?

Reply:
My bad, you are right. Replaced this with an assertion instead to generate a proper error.